modules/react/action-bar/lib/ActionBar.tsx

```diff
@@ -1,76 +1,37 @@
 import * as React from 'react';
-import {styled} from '@workday/canvas-kit-react/common';
-import {colors, commonColors, space, CSSProperties} from '@workday/canvas-kit-react/tokens';
+import {createComponent, styled, StyledType} from '@workday/canvas-kit-react/common';
+import {commonColors, colors, space} from '@workday/canvas-kit-react/tokens';
+import {HStack, HStackProps, StackSpacing} from '@workday/canvas-kit-labs-react';
 
-export interface ActionBarProps extends React.HTMLAttributes<HTMLDivElement> {
-  /**
-   * If true, fix the ActionBar to the bottom of the screen.
-   * @default false
-   */
-  fixed?: boolean;
+interface ActionBarProps extends Omit<HStackProps, 'spacing'> {
+  spacing?: StackSpacing;
 }
 
-function getFixedStyles(fixed = false): CSSProperties {
-  return fixed
-    ? {
-        position: 'fixed',
-        left: 0,
-        bottom: 0,
-        right: 0,
-      }
-    : {};
-}
-
-const ActionBarContainer = styled('div')<ActionBarProps>(
-  {
-    borderTop: `solid 1px ${colors.soap400}`,
-    background: commonColors.background,
+const ResponsiveHStack = styled(HStack)<HStackProps & StyledType>(({theme}) => ({
+  [theme.canvas.breakpoints.down('s')]: {
     padding: space.s,
-    boxShadow: '0 -2px 4px rgba(0, 0, 0, 0.08)',
-  },
-  ({fixed, theme}) => {
-    return {
-      ...getFixedStyles(fixed),
-      [theme.canvas.breakpoints.down('s')]: {
-        padding: space.xxs,
-      },
-    };
-  }
-);
-
-const ChildrenContainer = styled('div')(
-  {
-    display: 'inline-block',
-    padding: `0 ${space.m}`,
-    '*:not(:first-of-type)': {
-      marginLeft: space.s,
+    '> *:not(div)': {
+      flex: 1,
     },
   },
-  ({theme}) => ({
-    [theme.canvas.breakpoints.down('s')]: {
-      display: 'flex',
-      padding: space.xxs,
-      justifyContent: 'center',
-      flexDirection: 'row-reverse',
-      '> *': {
-        flex: 1,
-        '&:not(:first-of-type)': {
-          marginRight: space.s,
-          marginLeft: 0,
-        },
-      },
-    },
-  })
-);
-
-export default class ActionBar extends React.Component<ActionBarProps> {
-  public render() {
-    const {fixed, children, ...elemProps} = this.props;
+}));
 
-    return (
-      <ActionBarContainer {...elemProps} fixed={fixed}>
-        <ChildrenContainer>{children}</ChildrenContainer>
-      </ActionBarContainer>
-    );
-  }
-}
+export const ActionBar = createComponent('div')({
+  displayName: 'ActionBar',
+  Component: (elemProps: ActionBarProps, ref, Element) => (
+    <ResponsiveHStack
+      ref={ref}
+      as={Element}
+      spacing="s"
+      depth={1}
+      background={commonColors.background}
+      borderTop={`solid 1px ${colors.soap400}`}
+      padding={`${space.s} ${space.xl} `}
+      position="fixed"
+      bottom={0}
+      left={0}
```

Comment:
Worth noting that the `left` style prop is not RTL-safe. It will always be `left`. Is that what we want? Or should we use `insetInlineStart` which is RTL-safe? Here's some documentation on position props for reference: https://workday.github.io/canvas-kit/?path=/docs/labs-common-box--position

Reply:
I added alignment to the right side instead of setting full width for container. So now it should work correctly for both directions.

Reply:
Cool. Good to know. Thanks!